### DISCUSSION TITLE
Transaction Time

### DISCUSSION THREAD

**Discussion by 8BallBomBom** [stalwartlabs/smtp-server#50] [Q&A] [answered]
First off thank you for creating a great set of softwares to handle mail and get the job done quickly and easily.
Such a breeze compared to the last few setups.

I do have a question though.
If compared to postfix smtp the transaction times seem to be slow at least initially.
Is that intentional? and can i change anything to speed things up?

The time difference is as follows.
**Postfix:** 0.2 - 1.2 seconds.
**Stalwart:** 6 - 8 seconds.
Same server specs, same location, same tests.

**Comment by mdecimus** [comment]
Hi,

This is most likely related to a DNS issue. Stalwart checks the reverse IP address as well as SPF, DKIM and DMARC records. If there are issues performing these DNS queries this will delay the requests.

**Comment by 8BallBomBom** [comment]
Was thinking the same but the DNS is setup based on what Stalwart provides and the SMTP Banner has been altered to include the hostname and the Reverse DNS has been configured.

When running various tests via [this](https://mxtoolbox.com/supertool) tool.
Everything seems to pass if you don't include the slow SMTP Transaction time 🤔

**Reply by mdecimus** [reply, marked as answer]
> Was thinking the same but the DNS is setup based on what Stalwart provides

The default is to use the local resolver. If your resolver has delays this will impact the transaction times.

Postfix by default only uses the DNS to check the reverse IP address but does not perform any SPF, DKIM or DMARC checks and doesn't do SPAM filtering either. For these it requires external tools such as OpenDKIM, OpenDMARC, etc and Rspamd which, if you haven't configured, it won't be a fair comparison.
If you want to test the raw speed of Stalwart vs Postfix you'll have to disable DKIM, SPF, DMARC as well as the spam filters. Probably the reverse IP checks too if these are not enabled in Postfix.

**Reply by 8BallBomBom** [reply]
Alrighty that makes sense 😃
Thanks for the info.
